### ISSUE TITLE
Can we run Lidraughts bot in heroku

### ISSUE THREAD

**Issue by SathyaKarthik1212** [AttackingOrDefending/lidraughts-bot#3]
The title is the question

Please tell me @AttackingOrDefending

**Comment by littlebasiccoder** [comment]
yes but not to make it easier for non coders like you @SathyaKarthik1212

**Comment by AttackingOrDefending** [comment]
Yes. You do exactly what you would do for lichess-bot. I will make some changes explaining how to run a bot in heroku.

**Comment by AttackingOrDefending** [comment]
I pushed the changes explaining it. You can find them [here](https://github.com/AttackingOrDefending/lidraughts-bot/blob/main/heroku/README.md).

**Comment by SathyaKarthik1212** [comment]
@AttackingOrDefending what is the recommend engine to make bot using heroku

**Comment by AttackingOrDefending** [comment]
You can use Scan (https://github.com/rhalbersma/scan).

**Comment by NikitaRaik** [comment]
If I run this code in heroku, how can I use this for what?

**Comment by NikitaRaik** [comment]
And could you tell me how to use 8x8 CBEngine in detail?

**Comment by AttackingOrDefending** [comment]
Use the steps [here](https://github.com/AttackingOrDefending/lidraughts-bot/blob/main/heroku/README.md), to run a bot on heroku. You can't run a CBEngine in heroku because heroku uses Linux while the engines were compiled for windows.

**Comment by NikitaRaik** [comment]
Thank you for your response.
I ran the code in heroku.
Then how can I use application in heroku?
I can't load page after deploy this code on heroku.
Thank you.

**Comment by AttackingOrDefending** [comment]
For an engine you can download Scan ([hjetten.home.xs4all.nl/scan/scan.html](https://hjetten.home.xs4all.nl/scan/scan.html)). Copy or rename the file config.yml.default to config.yml. Create an account and get a bot token ([github.com/AttackingOrDefending/lidraughts-bot#lidraughts-oauth](https://github.com/AttackingOrDefending/lidraughts-bot#lidraughts-oauth)). To setup the engine follow the steps [github.com/AttackingOrDefending/lidraughts-bot#setup-engine](https://github.com/AttackingOrDefending/lidraughts-bot#setup-engine). Edit the variants (supported_variants) and time controls (supported_tc) from the config.yml (use # to disable certain ones).

For hosting it on heroku:
- Move all the files from "./heroku" to "."
- In startbot.sh, change "chmod +x ./engines/engine_name" to "chmod +x ./engines/scan_linux" (rename Scan to scan_linux and use the linux version)
- Follow the last 3 steps from [github.com/AttackingOrDefending/lidraughts-bot/tree/main/heroku](https://github.com/AttackingOrDefending/lidraughts-bot/tree/main/heroku)
- Maybe you can deploy from github instead of heroku git and put the token in the environment variable but I haven't tested it.

**Comment by NikitaRaik** [comment]
I already deployed the code on heroku.
After deploy on heroku, how to use it?
Thank you.

**Comment by AttackingOrDefending** [comment]
Go to Resources (https://dashboard.heroku.com/apps/<your-app-name>/resources), click "Edit dyno formation" and enable the app and click "Confirm".

**Comment by NikitaRaik** [comment]
I already done and I saw this log.
![Screenshot_13](https://user-images.githubusercontent.com/79733831/166191453-47f5630e-c5ce-4a3e-8d0c-423e1ee6f861.png)

I want to know how to use it.
Thank you.

**Comment by NikitaRaik** [comment]
I'm going to make a checkers game AI engine using Node js or Python.
But I'm a beginner and have poor algorithm knowledge.
Do you know where is checkers game AI engine made by Python or Node js?
Thank you.

**Comment by AttackingOrDefending** [comment]
The log means that it works.

Examples of a Python checkers engine:
- https://github.com/techwithtim/Python-Checkers-AI
- https://github.com/akalverboer/python_draughts_10x10
- https://github.com/damiha/CheckersBot

**Comment by NikitaRaik** [comment]
Could you tell me how to use this code after deploying on Heroku?
Thank you.

**Comment by AttackingOrDefending** [comment]
They aren't compatible with lidraughts-bot. You are going to have to modify them in order to connect them to lidraughts-bot. You can use the homemade protocol. Otherwise, you can use an engine that supports the hub or dxp protocol that you can find on the internet.

**Comment by NikitaRaik** [comment]
I used scan from (https://github.com/rhalbersma/scan).
Then there is no error when building.
Could you send me a screenshot of compatibility with lidraughts-bot?
Thank you.

**Comment by AttackingOrDefending** [comment]
None of them are compatible with lidraughts-bot. The last one is probably the easiest to adapt, since it already uses pydraughts. See [here](https://github.com/damiha/CheckersBot/blob/master/ai_engine.py) for the file with the engine logic.

**Comment by SriMethan** [comment]
Note : - Anyone can run a bot on Heroku

But,

Finally, Heroku doesn't have any free plans, but I guess they have some eco plans or basic plans which charges only low cost to run the bot on Heroku,

If someone needs another free alternative of Heroku take a look at [cloudcommunity/Free-Hosting](https://github.com/cloudcommunity/Free-Hosting) 

If someone prefers a paid alternative of Heroku take a look at [cloudcommunity/Cloud-Free-Tier-Comparison](https://github.com/cloudcommunity/Cloud-Free-Tier-Comparison)

**Comment by SriMethan** [comment]
Hey!

Guess what I found!

[render.com/docs/migrate-from-heroku](https://render.com/docs/migrate-from-heroku)

Completely free

Only for web not worker though

for migrating to vercel from heroku I think I found a video too if you want...

But that's only web too
